# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../minit/compile.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- ./Makefile.orig	2005-04-08 19:24:04.000000000 +0200
+++ ./Makefile	2005-04-11 15:03:26.000000000 +0200
@@ -46,7 +46,7 @@
 serdo: serdo.o
 
 shutdown: shutdown.o split.o openreadclose.o opendevconsole.o
-	$(DIET) $(CROSS)$(CC) $(LDFLAGS) -o shutdown $^
+	$(DIET) $(CROSS)$(CC) $(LDFLAGS) -o shutdown $^ $(LDLIBS)
 
 %.o: %.c
 	$(DIET) $(CROSS)$(CC) $(CFLAGS) -c $<
@@ -65,7 +65,7 @@
 	$(DIET) $(CROSS)$(CC) $(CFLAGS) -o $@ $^
 
 hard-reboot: hard-reboot.c
-	$(DIET) $(CROSS)$(CC) $(CFLAGS) -o $@ $^
+	$(DIET) $(CROSS)$(CC) $(CFLAGS) -o $@ $^ $(LDLIBS)
 
 write_proc: write_proc.c
 	$(DIET) $(CROSS)$(CC) $(CFLAGS) -o $@ $^
--- ./serdo.c.orig	2005-02-26 18:00:22.000000000 +0100
+++ ./serdo.c	2005-04-11 14:59:39.000000000 +0200
@@ -11,6 +11,8 @@
 char* envp[MAXENV+2];
 int envc;
 
+extern char **environ;
+
 int continueonerror;
 
 int envset(char* s) {
--- minit-0.10/minit.c.vanilla	2005-04-11 15:24:23.000000000 +0200
+++ minit-0.10/minit.c	2005-04-11 15:24:10.000000000 +0200
@@ -26,6 +26,8 @@
 #undef MALLOC_TEST
 #endif
 
+#undef MALLOC_TEST
+
 #ifdef MALLOC_TEST
 extern void* __libc_malloc(size_t size);
 extern void* __libc_realloc(void* x,size_t size);
